lib/pipeline.ts

```diff
@@ -319,16 +320,25 @@ Pipeline.prototype.exec = function(callback: CallbackFunction) {
           bufferMode = true;
         }
         if (bufferMode) {
-          data = Buffer.concat([
-            typeof data === "string" ? Buffer.from(data, "utf8") : data,
+          if (!buffers) {
+            buffers = [];
+          }
+          if (typeof data === "string") {
```

Comment:
Data is ignored if its already a buffer

Reply:
`data` cannot be a Buffer. I cleaned up the code a bit to make this clear.
Your comment made me aware that `buffers` is actually never re-set, so I did that, too.